### PR TITLE
Move yolox detector to examples

### PR DESCRIPTION
**Pull request by yushan111**
## Why this PR? 
This PR treats the yolox detector as a user plug-in detector (in examples), to 
* isolate yolox code dependencies from VQPy
* ease the future detection models built in (minimize dependencies).
## Changes in VQPy
 1. To install VQPy only, users don't need to install yolox and the pre-trained in `model` directory that we specified, or install the dependencies yolox introduced. 
 2. Add how to customize and register a yolox detector to VQPy in examples

## Future Plan
*  Refine detector interface and add built-in detectors

### CODE REVIEW

File: examples/list_red_moving_vehicle/yolox_detector.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+from yolox.data.data_augment import ValTransform
+from yolox.exp.build import get_exp
+from yolox.utils import postprocess
+from yolox.utils.model_utils import get_model_info
+
+
+def generate_register_yolox_detector(model_path=None):
+
+    class YOLOXDetector(DetectorBase):
```
Comment (borontion) [inline examples/list_red_moving_vehicle/yolox_detector.py:23]: Is there any specific reason to define `YOLOXDetector` as an inner class? Inner class code looks a little bit difficult to understand.

File: examples/list_red_moving_vehicle/yolox_detector.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+"""
+
+from typing import Dict, List
+
+import numpy as np
+import torch
+from loguru import logger
+from vqpy.base.detector import DetectorBase
+from vqpy.utils.classes import COCO_CLASSES
+from vqpy.detector.logger import register
```
Comment (borontion) [inline examples/list_red_moving_vehicle/yolox_detector.py:13]: I don't understand why the `register` function is under `logger`. @whzzt

File: examples/list_red_moving_vehicle/yolox_detector.py (hunk 1 above)
Comment (yushan111) [reply examples/list_red_moving_vehicle/yolox_detector.py]: It is for users to ingest `model_path` for their pretrained model (without changing the Detector interface).

I have updated code. In the newer version, users can specify a `detector_model_path` in `vqpy.launch`, and `model_path` is an instance attribute of Detector.  
```python
class YOLOXDetector(DetectorBase):

    def __init__(self, model_path, device="gpu", fp16=True):
          pass 

vqpy.launch(...,
     detector_model_path=args.yolox_model_path,
)
```
How about the updated interface?

Comment (borontion) [reply examples/list_red_moving_vehicle/yolox_detector.py]: Looks better

**Review by borontion — APPROVED** [review]

**Comment by yushan111** [comment]
Thanks for the review and suggestions! 
I will merge the PR to unblock the next-step development. And I will create a new PR to address any further comments.

**Pull request merged**